packages/api/tests/endpoints/test-granules.js

```diff
@@ -843,40 +852,132 @@ test.serial('move a granule with no .cmr.xml file', async (t) => {
+        filePgModel,
+        granuleModel,
+        granuleFileName,
+      });
 
-      newGranule.files = [
+      const destinationFilepath = `${process.env.stackName}/granules_moved`;
+      const destinations = [
         {
+          regex: '.*.txt$',
           bucket,
-          fileName: `${newGranule.granuleId}.txt`,
-          key: `${process.env.stackName}/original_filepath/${newGranule.granuleId}.txt`,
+          filepath: destinationFilepath,
         },
         {
-          bucket,
-          fileName: `${newGranule.granuleId}.md`,
-          key: `${process.env.stackName}/original_filepath/${newGranule.granuleId}.md`,
+          regex: '.*.md$',
+          bucket: thirdBucket,
+          filepath: destinationFilepath,
         },
         {
-          bucket: secondBucket,
-          fileName: `${newGranule.granuleId}.jpg`,
-          key: `${process.env.stackName}/original_filepath/${newGranule.granuleId}.jpg`,
+          regex: '.*.jpg$',
+          bucket,
+          filepath: destinationFilepath,
         },
       ];
 
-      await granuleModel.create(newGranule);
-
-      await Promise.all(
-        newGranule.files.map(
-          (file) =>
-            s3PutObject({
-              Bucket: file.bucket,
-              Key: file.key,
-              Body: 'test data',
-            })
-        )
+      const response = await request(app)
+        .put(`/granules/${newGranule.granuleId}`)
+        .set('Accept', 'application/json')
+        .set('Authorization', `Bearer ${jwtAuthToken}`)
+        .send({
+          action: 'move',
+          destinations,
+        })
+        .expect(200);
+
+      const body = response.body;
+      t.is(body.status, 'SUCCESS');
+      t.is(body.action, 'move');
+
+      // Validate S3 Objects are where they should be
+      const bucketObjects = await s3().listObjects({
+        Bucket: bucket,
+        Prefix: destinationFilepath,
+      }).promise();
+
+      t.is(bucketObjects.Contents.length, 2);
+      bucketObjects.Contents.forEach((item) => {
+        t.is(item.Key.indexOf(`${destinationFilepath}/${granuleFileName}`), 0);
+      });
+
+      const thirdBucketObjects = await s3().listObjects({
+        Bucket: thirdBucket,
+        Prefix: destinationFilepath,
+      }).promise();
+
+      t.is(thirdBucketObjects.Contents.length, 1);
+      t.is(thirdBucketObjects.Contents[0].Key, `${destinationFilepath}/${granuleFileName}.md`);
+
+      // check the granule in dynamoDb is updated and files are replaced
+      const updatedGranule = await granuleModel.get({ granuleId: newGranule.granuleId });
+
+      updatedGranule.files.forEach((file) => {
+        t.true(file.key.startsWith(`${destinationFilepath}/${granuleFileName}`));
+        const destination = destinations.find((dest) => file.fileName.match(dest.regex));
+        t.is(destination.bucket, file.bucket);
+      });
+
+      // check the granule in postgres is updated
+      const pgFiles = await getPostgresFilesInOrder(
```

Comment:
This could perhaps be better or more explicitly named, perhaps getPostgresFilesInTestGranuleOrder, although that's getting a bit long.   This is ordering the postgres files to match the test fixture `newGranules` , so as to make it possible to do iterable comparisons below.

Reply:
Would it be just as easy to sort `newGranule.files` and `pgFiles` by the same thing (`key`?) and then do a deepEqual comparison between two arrays? It's not necessary though